Reintroduce pageClasses to fix dark mode and fix eslint config

emmalewis opened the pull request:
- Reintroduces pageClasses to fix dark mode (issue [reported by Sam Learner](https://ft-editorial.slack.com/archives/C03CYK6Q1UN/p1697034747510389))
- In doing so I discovered that `eslint --fix` wasn't working correctly so this PR also fixes the configuration for that

aendra-rininsland reviewed (changes requested):
LGTM but you might be able to simply remove the whole settings.import/resolver stanza from .eslintrc as well as eslint-import-resolver-alias. Probably can also remove MDX support while you're at it, we don't have Storybook in Starter Kit anymore so having MDX support is unnecessary (you'll want to remove it also from "extends" in the eslintrc)

emmalewis commented:
Super, thank you. For the paper trail: I removed the mdx stuff. I realised that we actually did need the alias resolver specifically for the g-components styles (but not for g-components js components) so I updated the `.eslintrc` file to reflect that and left the eslint-import-resolver-alias.
